### PR TITLE
Reuse MixedDbClient to improve performance.

### PR DESCRIPTION
**Pull request by liuh-80**
Reuse MixedDbClient to improve performance.

#### Why I did it
Every API request will create a new MixedDbClient, also create ZMQ connection and table, which is not necessary.

#### How I did it
Add a cache to reuse MixedDbClient by ZMQ address.

#### How to verify it
Pass E2E test, create Dash resources correctly.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog
Reuse MixedDbClient to improve performance.

### CODE REVIEW

**Comment by ganglyu** [comment]
How to verify re-connect? Do you have unit test?

**Comment by liuh-80** [comment]
> How to verify re-connect? Do you have unit test?

Will create UT and update PR later.

**Comment by qiluo-msft** [comment]
> extern "C" {

We are moving away from manually writing c++ files in this repo, and prefer swig for new code. Could you try use SWIG?

---
Refers to: libcswsscommon/include/capi/zmqproducerstatetable.h:11 in 04aaaa2. [](commit_id = 04aaaa224752d6213b1cefab4584251ad48da390, deletion_comment = False)

File: sonic_data_client/mixed_db_client.go
Comment (qiluo-msft) [inline sonic_data_client/mixed_db_client.go]: We still need to access AppDB non-dash tables. Does this change impact the old feature?

Comment (liuh-80) [reply sonic_data_client/mixed_db_client.go]: @ganglyu , could you please confirm? As I know GNMI is only for Dash.

**Comment by liuh-80** [comment]
> > extern "C" {
> 
> We are moving away from manually writing c++ files in this repo, and prefer swig for new code. Could you try use SWIG?
> 
> Refers to: libcswsscommon/include/capi/zmqproducerstatetable.h:11 in [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390). [](commit_id = [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390), deletion_comment = False)

I will move all code change in c++ file to .go file, and use another PR to switch to swig.

File: sonic_data_client/mixed_db_client.go
Comment (ganglyu) [reply sonic_data_client/mixed_db_client.go]: We need to support dash tables and non-dash tables.

Comment (ganglyu) [inline sonic_data_client/mixed_db_client.go]: typo: getMexedDbClient should be Mixed

**Comment by ganglyu** [comment]
> > > extern "C" {
> > 
> > 
> > We are moving away from manually writing c++ files in this repo, and prefer swig for new code. Could you try use SWIG?
> > Refers to: libcswsscommon/include/capi/zmqproducerstatetable.h:11 in [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390). [](commit_id = [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390), deletion_comment = False)
> 
> I will move all code change in c++ file to .go file, and use another PR to switch to swig.
Please refer to:
https://github.com/sonic-net/sonic-gnmi/blob/master/swsscommon/Makefile

**Comment by liuh-80** [comment]
> > extern "C" {
> 
> We are moving away from manually writing c++ files in this repo, and prefer swig for new code. Could you try use SWIG?
> 
> Refers to: libcswsscommon/include/capi/zmqproducerstatetable.h:11 in [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390). [](commit_id = [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390), deletion_comment = False)

Fixed, switch to swig wrapper in another PR: https://github.com/nvidia-sonic/sonic-gnmi/pull/18

Will update this PR after that PR merge first.

File: sonic_data_client/mixed_db_client.go
Comment (liuh-80) [reply sonic_data_client/mixed_db_client.go]: Fixed
CUrrently the C++ class inheritance information missing in go wrapper, so create a new PR to update sonic-swss-common SWIG file in this PR: https://github.com/sonic-net/sonic-swss-common/pull/794

**Comment by liuh-80** [comment]
> > > extern "C" {
> > 
> > 
> > We are moving away from manually writing c++ files in this repo, and prefer swig for new code. Could you try use SWIG?
> > Refers to: libcswsscommon/include/capi/zmqproducerstatetable.h:11 in [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390). [](commit_id = [04aaaa2](https://github.com/nvidia-sonic/sonic-gnmi/commit/04aaaa224752d6213b1cefab4584251ad48da390), deletion_comment = False)
> 
> I will move all code change in c++ file to .go file, and use another PR to switch to swig.



> > How to verify re-connect? Do you have unit test?
> 
> Will create UT and update PR later.

Fixed, Add new UT for mixed_db_client

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by liuh-80** [comment]
This PR need wait for sonic-buildimage repo dash branch update sonic-swss-common submodule. will create PR and update to here later

**Comment by liuh-80** [comment]
WIll merge this PR after submodule update PR merge: https://github.com/nvidia-sonic/sonic-buildimage/pull/203

**Comment by liuh-80** [comment]
Will merge this PR because the sonic-swss-common submodule in dash sonic-buildimage repo currently is very old, the depends code change already not there before this PR.

**Pull request merged**
